tests/conftest.py

```diff
@@ -1,15 +1,52 @@
 from collections.abc import Generator
+from multiprocessing import Process
+from time import sleep
+from typing import Any
 
 import pytest
+import uvicorn
 from fastapi.testclient import TestClient
-from sqlmodel import Session, SQLModel
+from playwright.sync_api import sync_playwright
+from sqlmodel import Session
 from tad.core.config import settings
 from tad.core.db import get_engine
 from tad.main import app
-from tad.repositories.statuses import StatusesRepository
-from tad.repositories.tasks import TasksRepository
-from tad.services.statuses import StatusesService
-from tad.services.tasks import TasksService
+
+
+def run_server():
+    settings.APP_DATABASE_FILE = "database.sqlite3.test"
+    uvicorn.run(app, host="127.0.0.1", port=8000)
+
+
+def wait_for_server_ready(url: str, timeout: int = 30):
+    with sync_playwright() as p:
+        browser = p.chromium.launch()
+        page = browser.new_page()
+        for _ in range(timeout):
+            try:
+                page.goto(url)
+                browser.close()
+                return True  # noqa
+            except Exception as e:
+                print(e)
+                sleep(1)
+        browser.close()
+        raise Exception(f"Server at {url} did not become ready within {timeout} seconds")  # noqa: TRY003 TRY002
+
+
+@pytest.fixture(scope="module")
+def _start_server():
```

Comment:
`_start_server` -> `start_server`?

Reply:
I changed it to _ because I got this warning:
PT004 Fixture `start_server` does not return anything, add leading underscore

Reply:
Hm, I don't think I agree. We could instead maybe return a url + port?

Reply:
That sounds better, I wil do that.